### PR TITLE
Try catch for lib/index

### LINKED ISSUE #167: Exception handling for index functions
Add some level of exception handling(try-catch blocks) for functions inside `lib` folder.

### PR DESCRIPTION
**Pull request by VibhorCodecianGupta**
fixes #167 

@umeshp7 @shreys7 another attempt.
Please review whether things work this time around. Also, I found a lot of files inside `lib` of individual codegens which could use try-catch too. Shall we do it?

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -11,23 +11,28 @@ module.exports = {
...
-      var lang = codegen.lang.trim(),
-        currentVariant = codegen.variant.trim();
-      return language === lang.toLowerCase() && variant.toLowerCase() === currentVariant.toLowerCase();
-    });
+    try {
+      var validCodegen = languageMap.filter((codegen) => {
+        var lang = codegen.lang.trim(),
+          currentVariant = codegen.variant.trim();
+        return language === lang.toLowerCase() && variant.toLowerCase() === currentVariant.toLowerCase();
+      });
```
Comment (gamblegoon) [inline lib/index.js]: lib/index.js

Comment (gamblegoon) [reply lib/index.js]: thank you team for your point of directions

Comment (VibhorCodecianGupta) [reply lib/index.js]: Is there a review here? I didn't get you

**Review by webholik — COMMENTED** [review]
Passing errors in callback's first argument is standard pattern in NodeJs. This is also documented in the code-generators README. Throwing an exception should be reserved for situations where there is no scope for recovery. 

For `try catch` blocks we need to know which errors we are  expecting and how to recover when we get them. Just putting entire block of code in try and console logging the exception is not how it should be done. 

Since it's been a long time since this PR was created, I will be closing this one.

**Pull request closed without merging**